### ISSUE TITLE
Search feature in sketch list

### ISSUE THREAD

**Issue by yining1023** [processing/p5.js-web-editor#231]
I think it would be nice to have a search bar in the sketch list as an enhancement feature. Sometimes there are too many sketches in the list.

**Comment by catarak** [comment]
I think this would be a great feature!

**Comment by saumya1906** [comment]
I would like to take up this issue. Where should I ideally place the search bar in the sketch list overlay?

**Comment by catarak** [comment]
@saumya1906 have I added you to the Zeplin project? There's a mockup in there.

**Comment by saumya1906** [comment]
Katyayani added me to the Zeplin project.

**Comment by catarak** [comment]
<img width="513" alt="screen shot 2017-07-11 at 1 41 23 pm" src="https://user-images.githubusercontent.com/6063380/28081854-b685a068-663e-11e7-92bb-022166425b6e.png">
<img width="515" alt="screen shot 2017-07-11 at 1 41 32 pm" src="https://user-images.githubusercontent.com/6063380/28081855-b78d468c-663e-11e7-8ffe-5ec6440f9b0d.png">
<img width="515" alt="screen shot 2017-07-11 at 1 41 41 pm" src="https://user-images.githubusercontent.com/6063380/28081857-b8dbeffc-663e-11e7-81f6-17eb071496fe.png">

More details on the [Zeplin project](https://zpl.io/Z2cihf1).

**Comment by dhruvdutt** [comment]
@saumya1906 Any update about this?
@catarak Can I take this up? Also, please add me on Zeplin.

**Comment by catarak** [comment]
@dhruvdutt here's a link to the public Zeplin files: https://scene.zeplin.io/project/55f746c54a02e1e50e0632c3. If you need more details like pixel measurements and stuff i can add you to the full project!

**Comment by LakshSingla** [comment]
@catarak After the new implementation of the new SketchList layout, I would like to take up this issue. What all are the features one would like to search upon ?? :thinking:

**Comment by slowizzm** [comment]
I would like to be able to search by name, date, and an option to tag sketches and search by tag would be wonderful too.

**Comment by LakshSingla** [comment]
@slowizzm How do you propose searching by date ? Should there be some fixed format that the user should enter and it should filter the sketch by that. Alternatively, can you enumerate some sample queries (esp. pertaining to date, which could be entered in the search box).
Also, for the tags part, I believe it would require a change in the database schema and it makes sense to break it into a new issue.

**Comment by slowizzm** [comment]
After thinking about it, name and tags would be more than sufficient. When I said “date”, I was thinking about being able to search for date ranges corresponding with specific semesters. But, I think once the gallery is implemented, that will work better than searching by date.

So, my vote is a way to search by name and tag. If all else, by tag. 

Thanks.

**Comment by catarak** [comment]
i think date could be handled by being able to view sketches in ascending/descending order!

**Comment by ankitjena** [comment]
@LakshSingla are you working on this? Or am I free to work on it?

**Comment by LakshSingla** [comment]
I am not working on this issue, you can work on it. Although I believe that there is sorting functionality to be implemented after new SketchList design is implemented. Also regarding tags, there is a seperate discussion on #864, which is more evolved than this and you might like to read up on before starting to code.

**Comment by GaurangTandon** [comment]
@ankitjena Are you still working on this issue? Because I have gone through this issue in detail and would like to work on this.

**Comment by dhruvdutt** [comment]
I would like to work on this as well. ✋

**Comment by catarak** [comment]
okay, well, the first person to claim this ticket is working on it.

**Comment by dhruvdutt** [comment]
@ankitjena Hope you're working on this.

**Comment by catarak** [comment]
i'm thinking about adding one of those Github bots that will automatically assign users to an issue if they claim it and unassign them if they haven't completed it, to avoid situations like above.
